### PR TITLE
#902 Optimize code in dynamic data load init

### LINKED ISSUE #902: Improve the initialization implementation
In the current initialization implementation available into Dynawo, there are a few treatments that could be optimized because they are done on the same object, which is not fully utilized by the written code.

The idea would be to avoid searching into the structures several times when we are still working onto the same component.

### PR DESCRIPTION
**Pull request by lecourtoisflo**
fixes #902

### CODE REVIEW

**Comment by rosiereflo** [comment]
unit tests are not passing

File: dynawo/sources/Modeler/Common/DYNCompiler.cpp [hunk 1]
```diff
@@ -84,21 +84,21 @@ Compiler::compile() {
   getDDB();
 
   unitDynamicModelsMap_ = dyd_->getUnitDynamicModelsMap();
-  map<string, shared_ptr<ModelDescription> > blackboxes = dyd_->getBlackBoxModelDescriptions();
-  map<string, shared_ptr<ModelDescription> >::const_iterator itbbm;
+  const boost::unordered_map<string, shared_ptr<ModelDescription> >& blackboxes = dyd_->getBlackBoxModelDescriptions();
+  boost::unordered_map<string, shared_ptr<ModelDescription> >::const_iterator itbbm;
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/Common/DYNCompiler.cpp:88]: while we modify this part, please include the iterator into the for loop

Comment (rosiereflo) [reply dynawo/sources/Modeler/Common/DYNCompiler.cpp:88]: Also this means that the compilation will be done in a random manner doesn't this include non determinism in the logs? (it is OK in the compile log, but not in others)

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: dynawo/sources/Modeler/Common/DYNCompiler.cpp (hunk 1 above)
Comment (lecourtoisflo) [reply dynawo/sources/Modeler/Common/DYNCompiler.cpp]: OK

**Review by rosiereflo — APPROVED** [review]

**Review by gautierbureau — APPROVED** [review]

File: dynawo/sources/Modeler/Common/DYNDynamicData.cpp [hunk 2]
```diff
@@ -309,38 +280,57 @@ DynamicData::getParametersSet(const string& modelId, const string& parFile, cons
...
+    modelDescriptions_.insert(item);
+
+    if (mappedModel != NULL) {
+      mappedModel->insert(item);
+    }
   }
 
-  // @todo: instead of copying the full table one by one, directly copy the vector ?
+
+  // @todo: instead of copying the full table one by one, directly copy the vector ? => cannot be done by design of the iterators
```
Comment (adriguir) [inline dynawo/sources/Modeler/Common/DYNDynamicData.cpp:333]: IHMO, I wouldn't keep the todo but will only keep a comment like this "One by one copy due to the iterators design (no other possibility)"

**Review by adriguir — CHANGES_REQUESTED** [review]

**Review by adriguir — APPROVED** [review]

**Pull request merged**